### PR TITLE
Active record

### PR DESCRIPTION
**Pull request by DanSam5K**
In this project:
- We set up .githunb workflows linters  with rubocop and stylelint linters
- Then we created 3 models and associate the models after migrating them to the dabatases
- Then we added validation to our models and test them in the console
- Then Added proper documentation to the project
- Fix rubocop linter errors and made final touches.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Post < ApplicationRecord
+  belongs_to :author
+  belongs_to :comment
```
Comment (SajjadAhmad14) [inline app/models/post.rb]: It has_many:comments.

File: app/views/authors/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(author.errors.count, "error") %> prohibited this author from being saved:</h2>
+
+      <ul>
+        <% author.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (SajjadAhmad14) [inline app/views/authors/_form.html.erb:12]: [ OPTIONAL ] Move this logic to helper and render that helper here.

File: app/views/comments/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(comment.errors.count, "error") %> prohibited this comment from being saved:</h2>
+
+      <ul>
+        <% comment.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (SajjadAhmad14) [inline app/views/comments/_form.html.erb:12]: [ OPTIONAL ] Move this logic to helper and render that helper here.

File: app/views/posts/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(post.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+      <ul>
+        <% post.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (SajjadAhmad14) [inline app/views/posts/_form.html.erb:12]: [ OPTIONAL ] Move this logic to helper and render that helper here.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [x] README is professional and includes all the steps to run the project locally.
- [x] PR has a descriptive title and summary.
- [x] You have mastered Git flow.

## To Improve 🙏
- [x] Post has_many :comments.
- [x] A post can be created. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/creatingPost.png).
- [x] A comment can be created. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/user_has_many_comments.png).
- [x] Please see the inline comments for reference.
# Optional
- [ ] There should be no logic in views. You can move logic to helpers.
### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

File: app/models/post.rb (hunk 1 above)
Comment (DanSam5K) [reply app/models/post.rb]: Noted!

File: app/views/authors/_form.html.erb (hunk 2 above)
Comment (DanSam5K) [reply app/views/authors/_form.html.erb:12]: Alright Noted

File: app/views/comments/_form.html.erb (hunk 3 above)
Comment (DanSam5K) [reply app/views/comments/_form.html.erb:12]: Will do that sir!

File: app/views/posts/_form.html.erb (hunk 4 above)
Comment (DanSam5K) [reply app/views/posts/_form.html.erb:12]: Noted!

**Review by MkrtichSargsyan — APPROVED** [review]
## _Status:_ Approved  :hamster: :rocket:
Hello :wave:.

Awesome job on this project.:rocket: You have completed all the requested changes from the previous code reviewer.

You can go ahead and merge this branch to the base branch of this repo.
Good luck in your future projects and happy coding :relaxed:

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

**Pull request merged**